data/transactions/logic/assembler.go

```diff
@@ -2109,19 +2108,30 @@ func (ops *OpStream) warnf(format string, a ...interface{}) error {
 	return ops.warn(fmt.Errorf(format, a...))
 }
 
-// ReportProblems issues accumulated warnings and errors to stderr.
-func (ops *OpStream) ReportProblems(fname string) {
+// ReportProblems issues accumulated warnings and outputs errors to an io.Writer.
+func (ops *OpStream) ReportProblems(fname string, writer io.Writer) {
 	for i, e := range ops.Errors {
 		if i > 9 {
 			break
 		}
-		fmt.Fprintf(os.Stderr, "%s: %s\n", fname, e)
+		if fname == "" {
+			fmt.Fprintf(writer, "%s\n", e)
+		} else {
+			fmt.Fprintf(writer, "%s: %s\n", fname, e)
+		}
 	}
 	for i, w := range ops.Warnings {
 		if i > 9 {
 			break
 		}
-		fmt.Fprintf(os.Stderr, "%s: %s\n", fname, w)
+		if fname == "" {
+			fmt.Fprintf(writer, "%s\n", w)
+		} else {
+			fmt.Fprintf(writer, "%s: %s\n", fname, w)
+		}
+	}
+	if fname == "" {
+		fmt.Fprintf(writer, "%d errors", len(ops.Errors))
```

Comment:
I'd be ok with us not ever reporting the error count from in here.  It seems strange that we decide whether to do it based on if the fname is blank.  It's ok with me if the REST API just never says it.

Reply:
I agree, it looks funny in the code -- will delete!